cli/src/semgrep/rule_match.py

```diff
@@ -356,6 +356,18 @@ def get_end_line_hash(self) -> str:
         """
         last_line = self.get_individual_line(self.end.line)
         return hashlib.sha256(last_line.encode()).hexdigest()
+    
+    def is_direct(self) -> bool:
```

Comment:
Two comments:
1. `is_direct` it isn't really clear what `is_direct` means from the name. Probably we should have some `ScaMatch` subclass, but for now I'd maybe call it `is_sca_match_in_direct_dependency`. Verbose, but clear.
2. `"sca_info"` will not be present in `self.extra` if this wasn't an sca finding, so this should be checked before you access it

Reply:
Regarding the "sca_info" check, I believe it is performed in the `is_blocking()` method before I call the methods that use that field?

Reply:
I'm thinking about other people using this. If you have a method on a `RuleMatch` called `is_direct`, it's not at all clear that it might raise an exception.